src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -0,0 +1,153 @@
+        final DifferentialDrive m_drivetrain = new DifferentialDrive(m_leftMotors, m_rightMotors);
+        m_drivetrain.setSafetyEnabled(true);
+        m_drivetrain.setExpiration(0.1);
+        m_drivetrain.setMaxOutput(1.0);
+        m_drivetrain.setDeadband(0.05);
+
+        m_leftMotor_1.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+        m_leftMotor_2.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+        m_leftMotor_3.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+
+        m_rightMotor_1.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+        m_rightMotor_2.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+        m_rightMotor_3.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+        
+        // Encoder Helpers
+        // TODO: VERIFY CHANGES
+        DistanceSensor m_leftEncoderPosition = new DistanceSensor() {
+            public double getDistance() {
+                return (m_leftMotor_1.getSelectedSensorPosition() + m_leftMotor_2.getSelectedSensorPosition() + m_leftMotor_3.getSelectedSensorPosition())/3;
+                
+        }};
+        VelocitySensor m_leftEncodeVelocity = new VelocitySensor() {
+            public double getVelocity() {
+                return (m_leftMotor_1.getSelectedSensorVelocity() + m_leftMotor_2.getSelectedSensorVelocity() + m_leftMotor_3.getSelectedSensorVelocity())/3;
+            }
+        };
+
+        DistanceSensor m_rightEncoderPosition = new DistanceSensor() {
+            public double getDistance() {
+                return (m_rightMotor_1.getSelectedSensorPosition() + m_rightMotor_2.getSelectedSensorPosition() + m_rightMotor_3.getSelectedSensorPosition())/3;
+        }};
+        VelocitySensor m_rightEncodeVelocity = new VelocitySensor() {
+            public double getVelocity() {
+                return (m_rightMotor_1.getSelectedSensorVelocity() + m_rightMotor_2.getSelectedSensorVelocity() + m_rightMotor_3.getSelectedSensorVelocity())/3;
+            }
+        };
+
+        // Encoders
+        m_leftEncoderSensor = new BaseEncoderSensor(m_leftEncoderPosition, m_leftEncodeVelocity);
+        m_rightEncoderSensor = new BaseEncoderSensor(m_rightEncoderPosition, m_rightEncodeVelocity);
+        // NavX Gyro Initialization
+        Gyro m_gyro = new NavxGyro(m_navx);
+        
+        // Modular Drivetrain
+        m_modularDrivetrain = ModularDrivetrain.from(m_drivetrain);
+        
+        // Stability Module
+        m_stabilityModule = new StabilityModule(m_gyro, 0.05, 0.25);
+        m_stabilityModule.setTurnThreshold(0.075);
+
+        // Voltage Control Module 
+        m_voltageControlModule = new VoltageControlModule(DriveConstants.kNormalVoltage);
+
+        // Path Follower Module
+        m_pathFollowerModule = new PathFollowerModule(m_gyro, List.of(m_leftEncoderSensor, m_rightEncoderSensor), 0.5, 0.02); // TODO: Tune
+
+        //TODO: TARGET ALIGNMENT MODULE HERE
```

Comment:
The modules are defined here, but you will need to call the following to use them. I would also recommend using the power efficiency module (like this: https://github.com/Gongoliers/Robot2021/blob/6c3a15bcbdc719830e24fce6fd2d1584c0bac419/src/main/java/frc/robot/subsystems/Drivetrain.java#L88)

m_modularDrivetrain.setModules(m_stabilityModule, m_pathFollowerModule, m_voltageControlModule, m_powerEfficiencyModule)

Reply:
It might be good to start with just the voltage control and power efficiency module when testing things out though, then adding the others later